xdsl/dialects/arm/register.py

```diff
@@ -36,6 +38,27 @@ def infinite_register_prefix(cls):
         return "inf_"
 
 
+@irdl_attr_definition
+class NEONRegisterType(ARMRegisterType):
```

Comment:
My understanding of the current plan is for the registers to communicate the bitwidth of the operand/result, like in assembly. I'm not sure if we reached this conclusion the last time we spoke. We could also have just NEON registers and the operations carrying the bitwidths, but that might make operation definitions, assembly printing, and register allocation more complex.

Reply:
done, thank you

Reply:
We can always switch later, but I would start with bitwidths on the register for now